src/Map.php

```diff
@@ -36,6 +36,49 @@ public function __construct(array $map = [], LoggerInterface $logger = null) {
     }
 
     /**
+     * Merge this map with another map and return a new MapInterface
+     *
+     * Values in $map will be added to values in this instance.  Any number of additional maps may
+     * be passed to this method, i.e. $map->merge($map2, $map3, $map4, ...);
+     *
+     * @param MapInterface $map
+     * @return MapInterface
+     */
+    public function add(MapInterface $map) {
+        $combine = function($data, $map) {
+            foreach ($map as $key => $mask) {
+                $data[$key] = empty($data[$key]) ? $mask : ($data[$key] | $mask);
+            }
+            return $data;
+        };
+
+        $maps = array_slice((func_get_args()), 1);
```

Comment:
@jrbasso Darn, I read that calling slice on `func_get_args` would throw a warning in 5.4 without wrapping it, but I just tested it in 5.4, 5.5, and 5.6 and the information totally false.  I will remove.